src/sentry/runner/commands/devservices.py

```diff
@@ -553,3 +578,93 @@ def rm(project: str, services: list[str]) -> None:
             else:
                 click.secho("> Removing '%s' network" % network.name, err=True, fg="red")
                 network.remove()
+
+
+def check_health(service_name: str, containers: dict[str, Any]) -> None:
+    healthcheck = service_healthchecks.get(service_name)
+    if healthcheck is None:
+        return
+
+    click.secho(f"> Checking container health '{service_name}'", fg="yellow")
+
+    def hc() -> None:
+        healthcheck["check"](containers)
+
+    try:
+        run_with_retries(hc)
+    except subprocess.CalledProcessError:
+        click.secho(f"> '{service_name}' is not healthy", fg="red")
+        raise
+
+
+def run_with_retries(cmd: Callable[[], object], retries: int = 3, timeout: int = 5) -> None:
+    for retry in range(1, retries + 1):
+        try:
+            cmd()
+        except (subprocess.CalledProcessError):
+            if retry == retries:
+                raise
+            else:
+                click.secho(
+                    f"  > Health check failed, retrying in {timeout}s (attempt {retry+1} of {retries})...",
+                    fg="yellow",
+                )
+                time.sleep(timeout)
+        else:
+            return
+
+
+def check_kafka(containers: dict[str, Any]) -> None:
+    kafka_options = containers["kafka"]
+    zk_options = containers["zookeeper"]
+
+    zk_port = 2181
+    if zk_options["ports"]:
+        zk_port = zk_options["ports"][0]
+
+    subprocess.run(
+        (
+            "docker",
+            "exec",
+            kafka_options["name"],
+            "kafka-topics",
+            "--zookeeper",
+            # TODO: sentry_zookeeper:2181 doesn't work in CI, but 127.0.0.1 doesn't work locally
+            os.environ.get("ZK_HOST", f"{zk_options['name']}:{zk_port}"),
+            "--list",
+        ),
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+
+
+def check_postgres(containers: dict[str, Any]) -> None:
+    pg_options = containers["postgres"]
+    subprocess.run(
+        (
+            "docker",
+            "exec",
+            pg_options["name"],
+            "pg_isready",
+            "-U",
+            "postgres",
+        ),
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+
+
+class ServiceHealthcheck(TypedDict):
+    check: Callable[[dict[str, Any]], None]
```

Comment:
generally avoid TypedDict if you control all parts of the code -- this should be a `NamedTuple` or a concrete type

Reply:
I've created a class for all of the checks, please let me know if that works here or is there is a better/more preferable solution.